container download from shub / net crashes with silent loglevel

### Version of Singularity:
`3.7.1`
### Expected behavior

What did you expect to see when you do...?

`$ singularity  -s exec shub://repeatexplorer/repex_tarean:0.3.8.dbaa07f /bin/sh
`
should work without  crashing
### Actual behavior

Container download crashes because it does not expect nil progress bar callback at pull.go:100 : 

`$ singularity -s exec shub://repeatexplorer/repex_tarean:0.3.8.dbaa07f /bin/sh
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x55a1c13527d9]

goroutine 1 [running]:
github.com/sylabs/singularity/internal/pkg/client/shub.DownloadImage(0x55a1c186c460, 0xc000036068, 0xc0001fa000, 0x3cd, 0xc0004480e0, 0x1b, 0xc0005183e0, 0xd, 0xc0004481c0, 0x20, ...)
        github.com/sylabs/singularity@v0.0.0/internal/pkg/client/shub/pull.go:100 +0x859
github.com/sylabs/singularity/internal/pkg/client/shub.pull(0x55a1c186c460, 0xc000036068, 0xc00051c420, 0x0, 0x0, 0x7ffd75add097, 0x30, 0xc00071fb00, 0x0, 0x0, ...)
        github.com/sylabs/singularity@v0.0.0/internal/pkg/client/shub/pull.go:159 +0x513
github.com/sylabs/singularity/internal/pkg/client/shub.Pull(0x55a1c186c460, 0xc000036068, 0xc00051c420, 0x7ffd75add097, 0x30, 0xc00003c097, 0x1e, 0x0, 0x55a1c096ffa7, 0x3c, ...)
        github.com/sylabs/singularity@v0.0.0/internal/pkg/client/shub/pull.go:193 +0x9d
github.com/sylabs/singularity/cmd/internal/cli.handleShub(...)
        github.com/sylabs/singularity@v0.0.0/cmd/internal/cli/actions.go:86
github.com/sylabs/singularity/cmd/internal/cli.replaceURIWithImage(0x55a1c186c460, 0xc000036068, 0x55a1c1e69380, 0xc0002a03a0, 0x2, 0x2)
        github.com/sylabs/singularity@v0.0.0/cmd/internal/cli/actions.go:115 +0x4e5
github.com/sylabs/singularity/cmd/internal/cli.actionPreRun(0x55a1c1e69380, 0xc0002a03a0, 0x2, 0x2)
        github.com/sylabs/singularity@v0.0.0/cmd/internal/cli/actions.go:54 +0x11c
github.com/spf13/cobra.(*Command).execute(0x55a1c1e69380, 0xc0000320d0, 0x2, 0x2, 0x55a1c1e69380, 0xc0000320d0)
        github.com/spf13/cobra@v1.1.1/command.go:843 +0x4f9
github.com/spf13/cobra.(*Command).ExecuteC(0x55a1c1e74b40, 0x55a1c182a560, 0xc00018e480, 0xc0004dcec0)
        github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.1.1/command.go:895
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        github.com/spf13/cobra@v1.1.1/command.go:888
github.com/sylabs/singularity/cmd/internal/cli.ExecuteSingularity()
        github.com/sylabs/singularity@v0.0.0/cmd/internal/cli/singularity.go:430 +0x20b
main.main()
        github.com/sylabs/singularity@v0.0.0/cmd/singularity/cli.go:19 +0x52
`

<!-- if this is a feature request, you can ignore this next part -->

### Steps to reproduce this behavior

How can others reproduce this issue/problem?


### What OS/distro are you running

```
$ cat /etc/os-release


```


### How did you install Singularity

Write here how you installed Singularity. Eg. RPM, source.

--- a/internal/pkg/client/progress.go
+++ b/internal/pkg/client/progress.go
@@ -26,7 +26,10 @@ type ProgressCallback func(int64, io.Reader, io.Writer) error
 func ProgressBarCallback(ctx context.Context) ProgressCallback {
 
 	if sylog.GetLevel() <= -1 {
-		return nil
+		// If we don't need a bar visible, we just copy data through the callback func
+		return func(totalSize int64, r io.Reader, w io.Writer) error {
+			return CopyWithContext(ctx, w, r)
+		}
 	}
 
 	return func(totalSize int64, r io.Reader, w io.Writer) error {

--- a/internal/pkg/client/progress_test.go
+++ b/internal/pkg/client/progress_test.go
@@ -1,0 +1,52 @@
+// Copyright (c) 2018-2020, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package client
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/sylabs/singularity/pkg/sylog"
+)
+
+func TestProgressCallback(t *testing.T) {
+	const input = "Hello World!"
+	ctx := context.Background()
+
+	// Check the progress bar, or invisible copy-through, works at all sylog
+	// levels
+
+	levels := []int{
+		int(sylog.DebugLevel),
+		int(sylog.VerboseLevel),
+		int(sylog.InfoLevel),
+		int(sylog.WarnLevel),
+		int(sylog.ErrorLevel),
+		int(sylog.FatalLevel),
+	}
+
+	for _, l := range levels {
+		t.Run(fmt.Sprintf("level%d", l), func(t *testing.T) {
+			sylog.SetLevel(l, true)
+
+			cb := ProgressBarCallback(ctx)
+			src := bytes.NewBufferString(input)
+			dst := bytes.Buffer{}
+
+			err := cb(int64(len(input)), src, &dst)
+			if err != nil {
+				t.Errorf("Unexpected error from ProgressCallBack: %v", err)
+			}
+
+			output := dst.String()
+			if output != input {
+				t.Errorf("Output from callback '%s' != input '%s'", output, input)
+			}
+		})
+	}
+}

